test/e2e/instascale_machinepool_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	. "github.com/project-codeflare/codeflare-operator/test/support"
+)
+
+func TestInstascaleMachinePool(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+
+	if !IsOsd(test) {
+		test.T().Skip("Skipping test as not running on an OSD cluster")
+	}
+
+	namespace := test.NewTestNamespace()
+
+	// Test configuration
+	testConfigData := map[string][]byte{
```

Comment:
nit: Maybe that can be inlined and the `testConfigData` variable removed.

Reply:
Updated